Install and configure Symfony

MKodde opened the pull request:
The aim for this PR is to get a working Symfony installation that is tailored somewhat to our liking. Nothing is set in stone yet. I tried to follow the Symfony best practices and added the Ibuildings qa-pack to get a nice set of quality assurance tools.

Todo in new PRs:
- [x] Provide a Docker-based development environment
- [x] Expand the README.md
- [ ] Add a CHANGELOG.md and come up with a release strategy that fits our needs
- [x] Register the project in a CI tool of our choice (I have quite some Travis experience, but using Github Actions seems a nice option here)

MKodde committed:
Create .gitignore file

MKodde committed:
Install a basic Symfony application

A full Symfony 5 installation (for web) was created, components that are
likely not to be used have been removed.

MKodde committed:
Install and configure the qa-pack

In effect, the phplint, phpcs, phpmd, phpstan, deptrac, phpunit and
security checker have been enabled when the user runs `composer check`

In addition some PHPCS warnings have been fixed by phpcbf and manual
fixing.

MKodde committed:
Setup a Stepup namespace

MKodde committed:
Add README.md

MKodde commented:
Hi @sgomez Maybe I'm running ahead on this one, but here is a PR that sets up a Symfony 5 base installation. See the discription of the PR and the individual commit messages for more details. Feel free to give suggestions on how you'd like to see this setup take shape!

sgomez commented:
Why Symfony 5.0 and not Symfony 4.4 LTS? LTS 4.4 security support ends at November 2023 meanwhile standard 5.0 ends next July. Next LTS (5.4) won't be launch until November 2021.

After each standard version we will have 2 months only to migrate dependencies. Migration between LTS versions has a margin of 2 years.

sgomez commented on .gitignore:
```diff
@@ -0,0 +1,58 @@
+/web/build
+.env
+
+# OS generated files #
```
Just a comment of good practices with git. OS generated/IDE's files and directories like this and `.idea`  are better to be in a personal global ignore file. Because each developer has his preferences and this .gitignore file only should ignore things than belongs to the project.

MKodde replied:
Yes, good point. I copied this from my companies set of defaults. This is more of a defence in depth approach. As not all contributors might have their preferences set correctly in their personal git settings.

Seeing we will probably the only contributors for now I'm removing them.

MKodde committed:
Downgrade to a LTS version of Symofny

Using latest and greatest is tempting, but going for an LTS version of
Symfony is the way to go here.

Downgraded Symfony to 4.4.*

MKodde commented:
> Why Symfony 5.0 and not Symfony 4.4 LTS?

I fully agree! Downgraded to LTS.
